action.yml

```diff
@@ -60,11 +67,17 @@ runs:
       REPO: ${{ inputs.repo }}
       OSG_SERIES: ${{ inputs.osg_series }}
       REGISTRY: ${{ inputs.registry_url}}
+      TIMESTAMP_TAG: ${{ inputs.timestamp_tag }}
     shell: bash
     run: |
       docker_repo="$REGISTRY/${{ steps.generate-image-name.outputs.image }}"
       tag_list=()
-      TIMESTAMP=$(date +%Y%m%d-%H%M)
+      if [[ -n $TIMESTAMP_TAG ]]; then
+        # grab the YYYYMMDD-hhmm timestamp
+        TIMESTAMP=${TIMESTAMP_TAG: -13}
```

Comment:
As long as it's safe to assume that the `timestamp_tag` parameter ends in `YYYYMMDD-hhmm`, then it's fine to do it like you're doing it.

If the character preceding the date were, say, a dot instead of a hyphen, i would tend to write `TIMESTAMP=${TIMESTAMP_TAG##*.}` to strip of everything up to that character, but as it is, in `opensciencegrid/osgvo-docker-pilot3.6-release-YYYYMMDD-hhmm`, the preceding hyphen is neither the first nor the last hyphen in the string, so we can't do `${TIMESTAMP_TAG#*-}` or `${TIMESTAMP_TAG##*-}`.

You could also do something 'fancier' with regex capture groups to extract it if you really wanted, but what you have seems to be the most straightforward.

Reply:
Yeah, regex groups seemed more heavyweight than I wanted to go